utils/fix_rules.py

```diff
@@ -257,6 +286,77 @@ def rewrite_value_remove_prefix(line):
     return key + ": " + new_value
 
 
+class CCEFile:
+    def __init__(self, project_root=None):
+        if not project_root:
+            project_root = os.path.join(
+                os.path.dirname(os.path.abspath(__file__)), "..")
+        self.project_root = project_root
+
+    @property
+    def absolute_path(self):
+        raise NotImplementedError()
+
+    def line_to_cce(self, line):
+        return line
+
+    def line_isnt_cce(self, cce, line):
+        return line != cce
+
+    def read_cces(self):
+        with open(self.absolute_path, "r") as f:
+            cces = f.read().splitlines()
+        return cces
+
+    def remove_cce_from_file(self, cce):
+        file_lines = self.read_cces()
+        lines_except_cce = [
+            line for line in file_lines
+            if self.line_isnt_cce(cce, line)
+        ]
+        with open(self.absolute_path, "w") as f:
+            f.write("\n".join(lines_except_cce) + "\n")
+
+    def random_cce(self):
+        cces = self.read_cces()
+        random.shuffle(cces)
+        return cces[0].strip()
+
+
+class RedhatCCEFile(CCEFile):
+    @property
+    def absolute_path(self):
+        return os.path.join(self.project_root, "shared", "references", "cce-redhat-avail.txt")
+
+
+CCE_POOLS["redhat"] = RedhatCCEFile
+
+
+def add_to_the_section(file_contents, yaml_contents, section, new_keys):
+    to_insert = []
+
+    sec_ranges = find_section_lines(file_contents, section)
+    if len(sec_ranges) != 1:
+        raise RuntimeError("could not find one section: %s"
```

Comment:
I think it is ideal to also include the rule id in this and the assertion below. See the other helper methods where the path to the rule is explicitly passed. This makes it easier for the caller to inspect which particular file failed. My 2c. :-)

Reply:
The rule is not known here, but I have added the handling code to the upper levels.